sklearn/preprocessing/_discretization.py

```diff
@@ -210,6 +215,13 @@ def fit(self, X, y=None):
 
         n_samples, n_features = X.shape
 
+        valid_strategy = ("uniform", "quantile", "kmeans")
+        if self.strategy not in valid_strategy:
+            raise ValueError(
+                f"Valid options for 'strategy' are {valid_strategy}. "
+                f"Got strategy={self.strategy!r} instead."
+            )
+
```

Comment:
You can remove this part. This is taken care of by the `_parameter_constraints` above.